src/zen/websocket_server.cpp

```diff
@@ -488,6 +508,68 @@ class WsHandler
         return OK;
     }
 
+    int sendTopQualityCertificatesForScid(const std::string& scIdString, const std::string& clientRequestId)
+    {
+        uint256 scId;
+        scId.SetHex(scIdString);
+        
+        CCoinsViewCache &view = *pcoinsTip;
+        
+        UniValue mempoolTopQualityCert(UniValue::VOBJ);
+        UniValue chainTopQualityCert(UniValue::VOBJ);
+
+        {
+            LOCK(cs_main);
+            if (!view.HaveSidechain(scId)) {
+                LogPrint("ws", "%s():%d - sidechain id not found[%s]\n", __func__, __LINE__, scIdString);
+                return INVALID_PARAMETER;
+            }
+        
+            if (mempool.hasSidechainCertificate(scId))
+            {
+                const uint256& topQualCertHash = mempool.mapSidechains.at(scId).GetTopQualityCert()->second;
+                const CScCertificate& topQualCert = mempool.mapCertificate.at(topQualCertHash).GetCertificate();
+                const CAmount certFee = mempool.mapCertificate.at(topQualCertHash).GetFee();
+                CDataStream ssBlock(SER_NETWORK, PROTOCOL_VERSION);
+                ssBlock << topQualCert;
+                std::string certHex = HexStr(ssBlock.begin(), ssBlock.end());
+     
+                mempoolTopQualityCert.push_back(Pair("quality", topQualCert.quality));
+                mempoolTopQualityCert.push_back(Pair("certHash", topQualCertHash.GetHex()));
+                mempoolTopQualityCert.push_back(Pair("rawCertificateHex", certHex));
+                mempoolTopQualityCert.push_back(Pair("fee", FormatMoney(certFee)));
+            }
+
+            CSidechain sidechainInfo;
+
+            if (view.GetSidechain(scId, sidechainInfo)) {
+                const uint256& topQualCertHash = sidechainInfo.lastTopQualityCertHash;
+                const int topQualityCertQuality = sidechainInfo.lastTopQualityCertQuality;
+                CScCertificate topQualCert;
+
+                if (!topQualCertHash.IsNull()) {
+                    uint256 blockHash;
+                    if (GetCertificate(topQualCertHash, topQualCert, blockHash)) {
+                        CDataStream ssBlock(SER_NETWORK, PROTOCOL_VERSION);
+                        ssBlock << topQualCert;
+                        std::string certHex = HexStr(ssBlock.begin(), ssBlock.end());
+
+                        chainTopQualityCert.push_back(Pair("quality", topQualityCertQuality));
+                        chainTopQualityCert.push_back(Pair("certHash", topQualCertHash.GetHex()));
+                        chainTopQualityCert.push_back(Pair("rawCertificateHex", certHex));
```

Comment:
Just for symmetry with mempool, should we need the fee in the json obj we could use the view obj for getting it from the cert, for instance:
   certFee = topQualCert.GetFeeAmount(view.GetValueIn(topQualCert));